test/distrib/cpp/run_distrib_test_cmake_module_install.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+cd "$(dirname "$0")/../../.."
+
+echo "deb http://archive.debian.org/debian jessie-backports main" | tee /etc/apt/sources.list.d/jessie-backports.list
+echo 'Acquire::Check-Valid-Until "false";' > /etc/apt/apt.conf
+sed -i '/deb http:\/\/deb.debian.org\/debian jessie-updates main/d' /etc/apt/sources.list
+apt-get update
+apt-get install -t jessie-backports -y libssl-dev wget
+
+# Install CMake 3.16
+wget -q -O cmake-linux.sh https://github.com/Kitware/CMake/releases/download/v3.16.1/cmake-3.16.1-Linux-x86_64.sh
+sh cmake-linux.sh -- --skip-license --prefix=/usr
+rm cmake-linux.sh
+
+# Install gRPC and its dependencies
+mkdir -p "cmake/build"
+pushd "cmake/build"
+cmake \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DgRPC_INSTALL=ON \
+  -DgRPC_BUILD_TESTS=OFF \
+  -DgRPC_SSL_PROVIDER=package \
```

Comment:
FYI, I also tested this without `gRPC_SSL_PROVIDER=package` and it seems to work as well 
When set to `=module` it will just build everything with boringssl and ignore the system-wide libssl-dev (which might be a fair self-contained option for some users). With `=package` one can only use libssl-dev (OpenSSL) because boringssl doesn't provide any installation rules.

No need to change anything now (I'll merge as is), I just wanted to point out this observation (might come handy when writing the docs).